sql/types/set_test.go

```diff
@@ -172,6 +172,7 @@ func TestSetConvert(t *testing.T) {
 		{[]string{"one", "two"}, sql.Collation_utf8mb4_general_ci, "ONE", "one", false},
 		{[]string{"ONE", "two"}, sql.Collation_utf8mb4_general_ci, "one", "ONE", false},
 
+		{[]string{"one", "two"}, sql.Collation_Default, ",one,two", nil, true},
```

Comment:
Might be nice to add one more test entry for a non-error condition for a set that does allow the empty string (just because I noticed there's no coverage for that in these tests yet)